### PR TITLE
Make `workflow-trigger` an input to download artifact step

### PR DESCRIPTION
**Pull request by jukent**
I am unsure on why this step to trigger the build-build lives inside of the book-build. Let me know what you think.

E: Thanks @brian-rose for explaining that it doesn't trigger the trigger-book-build.yaml workflow, but tells the book-build action to look in that action for the artifact.

Related to https://github.com/ProjectPythia/projectpythia.github.io/pull/415 and https://github.com/ProjectPythia/projectpythia.github.io/issues/319

### CODE REVIEW

File: .github/workflows/build-book.yaml [hunk 1]
```diff
@@ -75,7 +75,6 @@ jobs:
         uses: dawidd6/action-download-artifact@v3.1.4
         with:
           github_token: ${{ secrets.GITHUB_TOKEN }}
-          workflow: trigger-book-build.yaml
```
Comment (brian-rose) [inline .github/workflows/build-book.yaml:78]: I think this line is needed to identify the workflow that produced the artifact.

Comment (jukent) [reply .github/workflows/build-book.yaml:78]: Okay, so it doesn't start a new action. On the portal site, this isn't a workflow though. The workflow that produces the artifact will be in `publish-site.yaml`. I'll try changing this to be an optional input with the trigger-book-build.yaml as its default value?

Comment (jukent) [reply .github/workflows/build-book.yaml:78]: That explains why the [error on my fork](https://github.com/jukent/projectpythia.github.io/actions/runs/8382046266/job/22955021220) say "Error: no matching workflow run found with any artifacts?"

Comment (jukent) [reply .github/workflows/build-book.yaml:78]: Any other questions I have about the inputs to this step can be answered in the [`action-download-artifact` documentation](https://github.com/dawidd6/action-download-artifact) 🤦‍♀️

**Review by brian-rose — APPROVED** [review]
This looks sensible to me @jukent !

**Pull request merged**